### PR TITLE
Reset Flipper instance before each test

### PR DESCRIPTION
**Pull request by mrodriguez90**
#### Board:
* [Ticket](https://www.notion.so/rootstrap/Configure-Flipper-for-testing-ef52eca64fcc4856801ec8bf13dfc627?pvs=4)
---
#### Description:

Configure Flipper for testing
---
#### Notes:
* Tests should start fresh with all features disabled.
---
#### Tasks:
- [x] Use Flipper `MemoryAdapter` in the test environment as recommended in the Flipper documentation.
- [x] Reset Flipper instance before each test, this means that all tests will start fresh with all features disabled
- [x] Some code demonstration in this PR that I'll remove before merging this. 
---
#### Risk:
* NO
---
#### Preview:
👁️ 

Screen Shot 2023-08-18 at 16 54 32
Screen Shot 2023-08-18 at 16 55 03

https://www.flippercloud.io/docs/testing

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -52,7 +52,11 @@ def self.from_social_provider(provider, user_params)
   def full_name
     return username if first_name.blank?
 
-    "#{first_name} #{last_name}"
+    if Flipper.enabled?(:test_feature)
+      "#{last_name} #{first_name}"
+    else
+      "#{first_name} #{last_name}"
+    end
   end
```
Comment (mrodriguez90) [inline app/models/user.rb:60]: I'll remove this change before merging the PR, this is just a small POC within the PR

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -60,6 +60,14 @@
...
       expect(user.full_name).to eq('John Doe')
     end
+
+    context 'when test_feature feature flag is on' do
+      before { Flipper.enable(:test_feature) }
+
+      it 'returns the last_name first' do
+        expect(user.full_name).to eq('Doe John')
+      end
+    end
```
Comment (mrodriguez90) [inline spec/models/user_spec.rb:70]: I'll remove this change before merging the PR, this is just a small POC within the PR

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -70,6 +78,7 @@
         expect {
           described_class.from_social_provider('provider', params)
         }.to change(described_class, :count).by(1)
+        expect(Flipper).not_to be_enabled(:test_feature)
```
Comment (mrodriguez90) [inline spec/models/user_spec.rb:81]: I'll remove this change before merging the PR, this is just a small POC within the PR

Comment (mrodriguez90) [reply spec/models/user_spec.rb:81]: This demonstrates that the feature flag does not save the state between different tests

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -42,6 +42,11 @@
   # Detects N+1 queries
   config.before { Prosopite.scan }
   config.after { Prosopite.finish }
+
+  # Reset previous flipper instance
+  config.before do
+    Flipper.instance = nil # Reset previous flipper instance
+  end
```
Comment (santib) [inline spec/rails_helper.rb:49]:
```suggestion
  config.before { Flipper.instance = nil }
```

**Review by santib — APPROVED** [review]
Great! I didn't know about this. Thanks for pointing out to the documentation

**Review by sebastiancaraballo — APPROVED** [review]

**Pull request merged**